cmd/rpcdaemon/commands/bor_snapshot.go

```diff
@@ -74,16 +74,28 @@ func (api *BorImpl) GetAuthor(number *rpc.BlockNumber) (*common.Address, error)
 
 	// Retrieve the requested block number (or current if none requested)
 	var header *types.Header
-	if number == nil || *number == rpc.LatestBlockNumber {
+	if blockNrOrHash == nil {
 		header = rawdb.ReadCurrentHeader(tx)
 	} else {
-		header, _ = getHeaderByNumber(ctx, *number, api, tx)
+		if blockNr, ok := blockNrOrHash.Number(); ok {
+			header, _ = getHeaderByNumber(ctx, blockNr, api, tx)
```

Comment:
Let's do an `if else` only here as done in bor and in `GetSnapshotProposer` below?